Update app for move2 and new moveVis features

robe2037 opened the pull request:
This PR updates the app to use move2 and exposes additional features available from moveVis.

A summary of the main updates is as follows:

- App no longer converts input `move2` objects to `moveStack`
- Enables all default `res` values from moveVis (`"mean"`, `"max"`, `"min"`, `"median"`)
- Enables all map types currently supported by the basemaps package and adds `map_token` setting to allow for access to API-protected basemaps (closes #3)
- Updates track coloring options:
    - Legend now matches the attribute used for track coloring rather than always using track ID (closes #10)
    - Adds `path_pal` setting to allow for adjustment of output color palette
    - Enables coloring by continuous attribute data (supported automatically by moveVis updates)
    - Removes empty linetype legend (fixes #12)
- Adds extent settings to allow user to manually control background map extent (closes #7)
    - This replaces the `margin_factor` setting for extent control (closes #13) and changes default equidistant argument to `FALSE` to prevent poor map dimensions that may result if tracks are highly skewed.
- Fixes codec errors that prevented mpeg and 3gp formats from rendering. This has to do with MoveApps system dependencies. mpeg files produce poor animations, so they are not recommended, but they should not fail. (fixes #8)
- Adds map attribution label to accommodate map tile user agreement citation requirements. Also adds the `hide_attribution` setting as a fallback for cases where the attribution does not fit on the map. The docs for the setting indicate that it is then the user’s responsibility to make sure the basemap tiles are cited. The logs produce the citation that is necessary for a given basemap as well.

Updates to moveVis also appear to resolve #14 , resolve #16, and resolve #6, as these issues are no longer reproducible in the new app. I suggest we can mark these as closed now and make new issues if they recur.

At the current time this app is based on a development version of moveVis found [here](https://github.com/robe2037/moveVis/tree/moveapps). Many of the changes have been proposed for PR into the main moveVis package repo, but since the timeframe for integrating these changes is unclear, it makes sense to build the app based on the dev version linked above for now. If these changes are later integrated into moveVis we can update the app to depend directly on moveVis and make any necessary updates at that time.

robe2037 committed:
Update color handling for dev moveVis

robe2037 committed:
Use moveVis argument names for clarity

robe2037 committed:
Prettify JSON

robe2037 committed:
Add more basemap options

Includes API-protected basemaps

robe2037 committed:
Update legend setting

robe2037 committed:
Add person

robe2037 committed:
Update dependencies

robe2037 committed:
Allow `res` to handle string values

robe2037 committed:
Tweaks

robe2037 committed:
Use `:` to in map types

robe2037 committed:
Avoid symbol interpretation in `set_units()`

robe2037 committed:
Remove unnecessary default

robe2037 committed:
Use 'colour' everywhere

robe2037 committed:
Add palette setting

Removes dependency on fields

robe2037 committed:
Update setting docs

robe2037 committed:
Update README.md

robe2037 committed:
Safely handle missing API key

robe2037 committed:
Update renv

robe2037 committed:
Expose `equidistant` setting

Sets default to `FALSE`

robe2037 committed:
Update `margin_factor` default

robe2037 committed:
Update README.md

robe2037 committed:
Add unit tests

Uses vdiffr for plot comparisons; refactors rFunction.R to enable testing of intermediate output

robe2037 committed:
Update renv.lock

robe2037 committed:
Force available codec for 3gp and mpeg formats; partial fix for #8

robe2037 committed:
Add map citation label

robe2037 committed:
Remove unsupported map types

robe2037 committed:
Add map tile citation URLs to logs

robe2037 committed:
Use explicit package calls

robe2037 committed:
Update README.md

robe2037 committed:
Treat int vars as qualitative

robe2037 committed:
Update moveVis dev package SHA in renv

robe2037 committed:
Replace `margin_factor` and `equidistant` with direct extent selection

robe2037 committed:
Fix ESRI attribution bug

robe2037 committed:
Logger tweaks

robe2037 committed:
Update README.md

robe2037 committed:
Remove specific moveVis dependency

robe2037 committed:
Link to map providers for example basemaps

More comprehensive and less maintenance than providing images

robe2037 committed:
Use non-token default map type config

robe2037 committed:
Remove unused data files

robe2037 committed:
Fix file path in RDS input test

Can now run with `test_dir()`

robe2037 committed:
Remove API token from sdk.R

Currently breaks when testing in Docker container

robe2037 committed:
Update dependencies

robe2037 committed:
Add system dependencies to Dockerfile

robe2037 committed:
Update dev moveVis location

robe2037 committed:
Always output map in CRS of input data

robe2037 committed:
Change template-sync branch

robe2037 committed:
Remove secrets env from .gitignore

robe2037 committed:
Fix appspec.json typo

robe2037 committed:
Fix empty `map_token` bug

robe2037 committed:
Add testthat dependency

This is really a dev dependency, but no good way to distinguish right now

robe2037 committed:
Only check validity of custom bbox, not default

robe2037 committed:
Improve missing attribute color var error

robe2037 committed:
Remove dependency on testthat

Expose `out_file` argument to devs, but not in settings

andreakoelzsch commented:
@annescharf @robe2037 Dear Anne and Finn, thanks so much for taking this up and doing the proper update to move2! Anne, I would also be happy if you move the ownership of this App to yourself or Finn. Is that an option? All the best, Andrea

robe2037 committed:
Move testthat.R to standard location

robe2037 committed:
Update renv.lock

robe2037 committed:
Clarify `res` memory limit error possibility

robe2037 commented:
@andreakoelzsch @annescharf

I've made some final tweaks to documentation, etc. but at this point it seems like the app is working on developer MoveApps from my testing. If either of you want to try to find any issues I may have missed or provide feedback on whether the app + setting documentation is intelligible I can make those modifications as well.

Happy to take ownership of the app!

robe2037 committed:
Use new README template

robe2037 committed:
Add custom `align_move_()` wrapper

Speeds up `align_move()` by subsetting data and setting smart default for `fill_na_values`

robe2037 committed:
Use `path_alpha = 1`

Improves look of track tails during animation

robe2037 committed:
Write frames to disk

robe2037 committed:
Revert "Write frames to disk"

This reverts commit 1d0d0a4b42e626f91a8d3f8cfc58fbabde533cbf.

robe2037 committed:
Use singular units specifications

"mins" not accepted despite plural being fine for other units.

robe2037 committed:
fix default unit value in appspec.json

robe2037 committed:
Use `interpolate = TRUE`

robe2037 committed:
Force Web Mercator output

annescharf committed:
update documentation

robe2037 committed:
Hide `map_res` setting

Behavior is not intuitive and default value is typically appropriate

robe2037 committed:
Add output dimension settings

robe2037 committed:
Force numeric `res` parameter

robe2037 committed:
Use `path_size = 2` default

Seems to be more appropriate for default map size

robe2037 committed:
Add param developer docs

Some app arguments are not exposed as settings and are documented here

robe2037 committed:
Add option to produce static test frame

robe2037 committed:
Use lat/lon in extent selection args/docs

robe2037 committed:
Update snapshots

robe2037 committed:
Use ggspatial for map annotation

robe2037 committed:
Additional settings/docs updates

robe2037 committed:
Add ggspatial dependency

robe2037 committed:
Log map extent in WGS84

robe2037 committed:
Use high res basemap tiles

robe2037 committed:
Tweak high-res docs

robe2037 committed:
Depend on basemaps explicitly

We need to install dev version from github; this ensures that the package shows up in MoveApps build file

robe2037 committed:
Update `dry_run` docs

robe2037 committed:
Docs

robe2037 committed:
Use latest moveVis

robe2037 committed:
Update basemap defaults

robe2037 committed:
Update snapshots

robe2037 committed:
Fix extent logger info

robe2037 committed:
Update renv.lock

robe2037 committed:
Docs
